### PR TITLE
feat(Dropdown): Add portal to Dropdown component

### PR DESCRIPTION
**Pull request by Artikodin**
- [x] Feature
- [ ] Fix
- [ ] Enhancement

## Description

We need to move the `<Dropdown />` component into a portal. I use the `<Portal>` component provided by the `react-portal` library to render the portal. The portal appear at the bottom of the body and it positioned according to the input.

## Todo - Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.

### CODE REVIEW

File: src/Dropdown/elements.js [hunk 1]
```diff
@@ -56,11 +56,11 @@ export const Header = styled.button`
 
 export const Menu = styled.ul`
   position: absolute;
-  top: calc(100% + 0.4rem);
-  left: 0;
+  top: ${({ position: { top } }) => `${top}px`};
+  left: ${({ position: { left } }) => `${left}px`};
   z-index: 1;
```
Comment (spartDev) [inline src/Dropdown/elements.js]: You should remove the `z-index` since we use `portal` now

**Comment by spartDev** [comment]
React portal now can manage when you click outside the portal. https://www.npmjs.com/package/react-portal#portalwithstate
Try to use this instead of onclickoutside

**Comment by Artikodin** [comment]
> onclickoutside

**Comment by Artikodin** [comment]
> React portal now can manage when you click outside the portal. https://www.npmjs.com/package/react-portal#portalwithstate
> Try to use this instead of onclickoutside

@spartDev I look around on the web and apparently the react-portal package doesn't allow you to control the state of is component check this [issue](https://github.com/tajo/react-portal/issues/181). 
So I've used the legacy onclickoutside but I add the class `ignore-react-onclickoutside` (see [doc](https://github.com/Pomax/react-onclickoutside#marking-elements-as-skip-over-this-one-during-the-event-loop)) to my portal to avoid the close event on it.

**Pull request merged**